tests/utils/managed_process.py

```diff
@@ -154,48 +168,71 @@ def _start_process(self):
             self.working_dir or os.getcwd(),
         )
 
-        stdin = subprocess.DEVNULL
-        stdout = subprocess.PIPE
-        stderr = subprocess.STDOUT
-
-        if self.display_output:
-            self.proc = subprocess.Popen(
-                self.command,
-                env=self.env or os.environ.copy(),
-                cwd=self.working_dir,
-                stdin=stdin,
-                stdout=stdout,
-                stderr=stderr,
-                start_new_session=True,  # Isolate process group to prevent kill 0 from affecting parent
-            )
-            self._sed_proc = subprocess.Popen(
-                ["sed", "-u", f"s/^/[{self._command_name.upper()}] /"],
-                stdin=self.proc.stdout,
-                stdout=subprocess.PIPE,
-            )
-
-            self._tee_proc = subprocess.Popen(
-                ["tee", self._log_path], stdin=self._sed_proc.stdout
-            )
+        # Single subprocess with unified output handling
+        self._proc = subprocess.Popen(
+            self.command,
+            env=self.env or os.environ.copy(),
+            cwd=self.working_dir,
+            stdin=subprocess.DEVNULL,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            start_new_session=True,  # Isolate process group to prevent kill 0 from affecting parent
+            text=True,  # Use text mode for easier line processing
+        )
 
-        else:
-            with open(self._log_path, "w", encoding="utf-8") as f:
-                self.proc = subprocess.Popen(
-                    self.command,
-                    env=self.env or os.environ.copy(),
-                    cwd=self.working_dir,
-                    stdin=stdin,
-                    stdout=stdout,
-                    stderr=stderr,
-                    start_new_session=True,  # Isolate process group to prevent kill 0 from affecting parent
-                )
-
-                self._sed_proc = subprocess.Popen(
-                    ["sed", "-u", f"s/^/[{self._command_name.upper()}] /"],
-                    stdin=self.proc.stdout,
-                    stdout=f,
-                )
-            self._tee_proc = None
+        # Single output processing function for both cases
+        def process_output():
+            try:
+                # Clear log file first
+                with open(self._log_path, "w", encoding="utf-8") as f:
+                    pass
```

Comment:
I mean... I guess this works. Somehow this doesn't feel right but I can't put my finger on it. The alternative is to use [`os.remove()`](https://docs.python.org/3/library/os.html#os.remove) which seems more intentional than this.